drivers/FileHandle.h

```diff
@@ -25,62 +25,85 @@ namespace mbed {
 /** \addtogroup drivers */
 /** @{*/
 
-/** An OO equivalent of the internal FILEHANDLE variable
+
+/** Class FileHandle
+ *
+ *  An OO equivalent of the internal FILEHANDLE variable
  *  and associated _sys_* functions.
  *
- * FileHandle is an abstract class, needing at least sys_write and
+ *  FileHandle is an abstract class, needing at least sys_write and
```

Comment:
Actually, technically, do you even need read() and write()? You could provide just one. Or maybe you only have len()?

The FileHandle is in essence so abstract that pretty much everything is device-dependent. In POSIX there are rules about classes of devices (real files, pipes, sockets...), but to a large extent devices can do pretty much what they like.

Then it's more a matter of common sense what sort of application you attach to what sort of device. (Don't run a database on a serial port, don't seek if you want to support tape drives, don't try to talk PPP to a text file).